### PR TITLE
Remove unused method DynamicTransformInitialInfo::operator==

### PR DESCRIPTION
**Pull request by jacobhinkle**
I just noticed a typo in this signature and decided to fix it, then realized it's probably unused so I figured I'd just remove it.

### CODE REVIEW

**Comment by jacobhinkle** [comment]
!build

File: csrc/dynamic_transform.h [hunk 1]
```diff
@@ -31,12 +31,6 @@ class DynamicTransformInitialInfoBuilder;
 //! sizes
 class DynamicTransformInitialInfo {
  public:
-  bool operator==(const DynamicTransformConcretizationInfo& other) const;
```
Comment (naoyam) [inline csrc/dynamic_transform.h:34]: Shouldn't the corresponding implementation be removed as well?

Comment (naoyam) [reply csrc/dynamic_transform.h:34]: Looks like it doesn't exist. 😃

**Review by naoyam — APPROVED** [review]

File: csrc/dynamic_transform.h (hunk 1 above)
Comment (jacobhinkle) [reply csrc/dynamic_transform.h:34]: :-D Oops!

**Pull request merged**